rust/processor/src/worker.rs

```diff
@@ -60,6 +69,11 @@ const MAX_RESPONSE_SIZE: usize = 1024 * 1024 * 20;
 const RECONNECTION_MAX_RETRIES: u64 = 100;
 // Consumer thread will wait X seconds before panicking if it doesn't receive any data
 const CONSUMER_THREAD_TIMEOUT_IN_SECS: u64 = 60 * 5;
+// Number of batches processed before gap detected
+const GAP_DETECTION_BATCH_COUNT: u64 = 50;
```

Comment:
this is potentially 50,000 transactions, right? Maybe it'd make sense to have a time based component to the check too

Reply:
what if it's processing large transactions that takes long time to insert to DB? it might panic and crash loop forever. having the detection based on # of transactions means if there's a gap it will eventually detect it and panic.

Reply:
we could do both- we just want progress, but 50k transactions is a _very_ long time; and it shouldn't take more than 30s for any gap, right? I think it's a question of how we can guarantee we panic in the lowest time possible

Reply:
We could test with panicking after 30s without any advancement of the gap counter

Reply:
Actually if we had a 30s counter, this would panic in the case of processing Graffio txn's. For example, we see it happening whenever we backfill a mainnet processor. When it hits the Graffio txn's, it becomes slower than 30s, but the processor is still working. We don't want it to panic or it'd be in a crash loop forever.